pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
Still strikes me as a little arbitrary given that "contextdir" is part of the input source definition.  You could argue the directory (as fetched from git or whatever) is already setup and the strategy should be responsible for using the contextdir to ensure it processes the right inputs (dockerfile name, pom.xml filename, etc) within that directory, but ok.